addon/components/link-to-component.js

```diff
@@ -15,6 +16,11 @@ export default LinkComponent.extend({
 
     let owner = getOwner(this);
 
+    assert(
+      `You attempted to use a link-to within a routeless engine, this is not supported.`,
```

Comment:
:+1: Seems reasonable to me, though the assertion should make it useless on its own (as it causes an exception).